### PR TITLE
qcbor: add new version 1.4.1

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **qcbor/1.4.1**

#### Motivation
There are lots of improvements in 1.4.1.

#### Details
https://github.com/laurencelundblade/QCBOR/compare/v1.3...v1.4.1

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ # To avoid need for it #define QCBOR_DISABLE_FLOAT_HW_USE
+ if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+@@ -88,6 +92,9 @@ set_target_properties(
+ include(GNUInstallDirs)
+ install(
+     TARGETS qcbor
++    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
++    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
++    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+     PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/qcbor"
```
Comment (AbrilRBS) [inline recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch:28]: The include folder is different in the new version compared to the old one, it's a bit suprising that no recipe change was needed 🤔 any insight into this @toge?

**Review by AbrilRBS — COMMENTED** [review]
Thanks! Minor question regarding the new include install for public headers, otherwise looks good :)

File: recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+         $<$<BOOL:${QCBOR_OPT_DISABLE_FLOAT_ALL}>:USEFULBUF_DISABLE_ALL_FLOAT>
+ )
+ 
+-if (BUILD_SHARED_LIBS)
++if (0)
+     target_compile_options(qcbor PRIVATE -Os -fPIC)
+ endif()
+ 
++set_target_properties(qcbor PROPERTIES
++    WINDOWS_EXPORT_ALL_SYMBOLS ON
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch]: @AbrilRBS 
Sorry, I missed this.
I will fix supporting shared build on Windows.

File: recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch (hunk 1 above)
Comment (toge) [reply recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch]: @AbrilRBS 
qcbor/1.4.0 seems to define public header files in CMakeLists.txt. 

https://github.com/laurencelundblade/QCBOR/commit/3e0bdc5b5d22192b5a6a8ce14269f77cd6b2bdc9

File: recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (toge) [reply recipes/qcbor/all/patches/1.4-0001-fix-cmake.patch]: @AbrilRBS 
Fixed.

**Review by AbrilRBS — DISMISSED** [review]
Thanks!

**Review by ErniGH — DISMISSED** [review]

**Review by ErniGH — APPROVED** [review]

**Pull request merged**